python/ray/dag/compiled_dag_node.py

```diff
@@ -66,27 +66,32 @@ def do_allocate_channel(
     self,
     reader_and_node_list: List[Tuple["ray.actor.ActorHandle", str]],
     typ: ChannelOutputType,
+    read_by_adag_driver: bool,
```

Comment:
Is this equivalent to `is_adag_output_channel`? Should we use that name which seems easier to follow

Can we have a default value = False as most of the cases would be false

Reply:
I think it is better not having default as it is kind of important flag that can cause hangs if set incorrectly. 

Regarding the name, I thought `is_adag_output_channel` exposes more implementation detail than this (that read by driver == output), and that was the original name before I changed it to this one. I don't have strong preference. Lmk if you want me to change.

Reply:
The problem with `adag_driver` is it's not a well defined concept so you need to explain it everywhere and can be confused with ray driver. I don't have a good idea right now, so will leave it to you.

Reply:
Hmm imo it is a reasonable wording, so unless there's strong pushback, I will probably just keep it